src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -77,35 +83,58 @@ export class DataSourceSelectable extends React.Component<
 
   async componentDidMount() {
     this._isMounted = true;
+    let filteredDataSources: Array<SavedObject<DataSourceAttributes>> = [];
+    let dataSourceOptions: DataSourceOption[] = [];
     getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
```

Comment:
could we covert Promise.then to await syntax for better readability and maintainability

Reply:
```suggestion
const ds = await  getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type']);
```